ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -5669,6 +5723,11 @@ static void ggml_vk_op_f32(ggml_backend_vk_context * ctx, vk_context& subctx, co
         // im2col uses only src1 and dst buffers
         ggml_vk_sync_buffers(subctx);
         ggml_vk_dispatch_pipeline(ctx, subctx, pipeline, { vk_subbuffer{ d_Y, y_buf_offset, y_sz }, vk_subbuffer{ d_D, d_buf_offset, d_sz } }, sizeof(PC), &pc, elements);
+    } else if (op == GGML_OP_COUNT_EQUAL) {
+        ggml_vk_sync_buffers(subctx);
+        // count_equal assumes that destination buffer is initialized with zeroes
+        ggml_vk_buffer_memset_async(subctx, d_D, d_buf_offset, 0, d_sz);
```

Comment:
There should either be a ggml_vk_sync_buffers in between, or you could make the memset skip the first 4B and they can run in parallel.

Reply:
Err, I guess you can't skip the first 4B if it needs to be cleared to zero. So it should do a sync.

Reply:
Yeah, there needs to be a sync inbetween the memset and the dispatch.

Reply:
Done